oxidation/bindings/generator/templates/binding_web_meths.rs.j2

```diff
@@ -0,0 +1,395 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+{# Macros -#}
+
+{#-
+ # Types conversions macros
+ #}
+
+
+{%- macro function_param_type(type) -%}
+{%- if type.kind == "ref" -%}
+{{ function_param_type(type.elem) }}
+{%- elif type.kind == "ref" -%}
+&{{ function_param_type(type.elem) }}
+{%- elif type.kind == "list" -%}
+Vec<{{ function_param_type(type.elem) }}>
+{%- elif type.kind == "optional" -%}
+Option<{{ function_param_type(type.elem) }}>
+{%- elif type.kind == "bool" -%}
+bool
+{%- elif type.kind == "int" -%}
+i64
+{%- elif type.kind == "float" -%}
+f64
+{%- elif type.kind in ("str", "str_based") -%}
+String
+{%- elif type.kind == "bytes" -%}
+Uint8Array
+{%- elif type.kind == "struct" -%}
+Object
+{%- elif type.kind == "variant" -%}
+Object
+{%- elif type.kind == "result" -%}
+Object
+{%- else -%}
+{{ raise("Unsupported type %r" % type) }}
+{%- endif -%}
+{%- endmacro -%}
+
+
+{%- macro render_function_param_cooking(param_name, type) -%}
+{% if type.kind == "ref" %}
+{{ render_function_param_cooking(param_name, type.elem) }}
+{% elif type.kind == "optional" %}
+let {{ param_name }} = match {{ param_name }} {
+    Some({{ param_name }}) => {
+        {{ render_function_param_cooking(param_name, type.elem) | indent(8) }}
+        Some({{ param_name }})
+    }
+    None => None,
+};
+{% elif type.kind == "str_based" %}
+let {{ param_name }} = match {{ param_name }}.parse() {
+    Ok(val) => val,
+    Err(_) => {
+        return Err(JsValue::from(TypeError::new("Not a valid {{ type.name }}")));
+    }
+};
+{% elif type.kind in ("struct", "variant", "result") %}
+let {{ param_name }} = {{ param_name }}.into();
+let {{ param_name }} = {{ render_js_value_to_rs(param_name, type) }};
+{% elif type.kind == "list" %}
+let mut {{ param_name }}_converted = Vec::with_capacity({{ param_name }}.len());
+for x in {{ param_name }}.iter() {
+    {{ render_function_param_cooking("x", type.elem) | indent }}
+    {{ param_name }}_converted.push(x);
+}
+let {{ param_name }} = {{ param_name }}_converted;
```

Comment:
```suggestion
 let {{ param_name }} = {{ param_name }}
     .iter()
     .map(|x| {
         {{ render_function_param_cooking("x", type.elem) | ident }}
     })
     .collect::<{{ type.elem }}>()
```

Reply:
Same as above: flat is better than nested here ^^